tasks/src/tasks/authenticate.rs

```diff
@@ -67,27 +92,47 @@ impl Task for AuthenticateTask {
 
     fn process_continuation_request_authenticate(
         &mut self,
-        _: CommandContinuationRequest<'static>,
+        continuation: CommandContinuationRequest<'static>,
     ) -> Result<AuthenticateData<'static>, CommandContinuationRequest<'static>> {
-        if self.ir {
-            Ok(AuthenticateData::Cancel)
-        } else if let Some(line) = self.line.take() {
-            Ok(AuthenticateData::r#continue(line))
+        let data = if self.ir {
+            if let AuthMechanism::XOAuth2 = self.mechanism {
+                // The current continuation request indicates an error,
+                // so we need to send an empty response in order to receive
+                // the final authentication error later on.
+                match continuation {
+                    CommandContinuationRequest::Basic(basic) => {
+                        let text = basic.text();
+                        error!("cannot authenticate using XOAUTH2 mechanism: {text}");
```

Comment:
We could deduplicate line 105 and 109 by putting the log after the match. I'm also not sure if we want to emit an `error!` or a `warn!`ing. An `error!` seems to ask for a an action. But we can't do anything here.

Reply:
Ah, I see why you did it this way :+1: